### PR TITLE
New Solar System Cheat Sheet

### PR DESCRIPTION
**Pull request by vikashvverma**
**What does your Instant Answer do?**

Displays basic information of our solar system in response to searches like `solar system cheat sheet`, 'planets cheatsheet` etc.

**What is the data source for your Instant Answer? (Provide a link if possible)**

[ https://en.wikipedia.org/wiki/Solar_System](https://en.wikipedia.org/wiki/Solar_System)

**Which communities will this Instant Answer be especially useful for? (gamer, book lovers, etc)**

All People

**Is this Instant Answer connected to a DuckDuckHack Instant Answer idea?**

No

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**

screen shot 2016-02-04 at 10 45 23 pm

IA Page : https://duck.co/ia/view/solar_system_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@vikashvverma Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

While the team and the community review your submission, we'd like to make sure you look good! If you'd like a free DuckDuckGo tshirt, please send over the following info to merch@duckduckgo.com, and be sure to put "Dev Merch" in the subject line:

Name:
Street Address:
City:
State / Province:
Country:
Phone (Non-US Orders only): 
T-shirt size:
GitHub username:
Link to this Pull Request:

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by vikashvverma** [comment]
@moollaza Can you review this?

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=solar system cheatsheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/solar_system_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=solar system cheatsheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/solar_system_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/solar_system.json [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        "Mars",
+        "Jupiter",
+        "Saturn",
+        "Uranus",
+        "Neptune"
+    ],
+    "sections": {
+        "Sun": [
+            {
+                "key": "Distance (earth)",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/solar_system.json]: @vikashvverma  Lets change this to "Distance from earth"

File: share/goodie/cheat_sheets/json/solar_system.json [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+                "key": "Mass",
+                "val": "1.0243×10^26 kg"
+            },
+            {
+                "key": "Temperature(mean)",
+                "val": "72 K"
+            },
+            {
+                "key": "Volume",
+                "val": "6.254×10^13 km^3 "
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/solar_system.json]: @vikashvverma lets remove the extra space at end of val field.
check other fields too.

Comment (vikashvverma) [inline share/goodie/cheat_sheets/json/solar_system.json]: That does not make any difference.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=solar system cheatsheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/solar_system_cheat_sheet_mobile.png)

**Comment by khflab** [comment]
Hey, i believe in a more simple description, like: 'Basic information about our solar system'.

Just my opinion, nice work.

**Comment by vikashvverma** [comment]
@khflab right man.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=solar system cheatsheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/solar_system_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/solar_system.json [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+        "Mars",
+        "Jupiter",
+        "Saturn",
+        "Uranus",
+        "Neptune"
+    ],
+    "sections": {
+        "Sun": [
+            {
+                "key": "Distance from earth",
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/solar_system.json]: Isn't this redundant as it's the same as the distance the earth is from the sun?

File: share/goodie/cheat_sheets/json/solar_system.json [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+                "val": "149,597,900 km"
+            },
+            {
+                "key": "Mass",
+                "val": "1.989 x 10^30 kg"
+            },
+            {
+                "key": "Temperature(center)",
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/solar_system.json]: Can you change this to "Core Temperature"

Comment (mintsoft) [inline share/goodie/cheat_sheets/json/solar_system.json]: Or `Temperature (core)` if you'd prefer to keep temperature at the start of the key

File: share/goodie/cheat_sheets/json/solar_system.json [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+            {
+                "key": "Distance from Sun (minimum/maximum)",
+                "val": "107,477,000/108,939,000 km"
+            },
+            {
+                "key": "Mass",
+                "val": "4.8675×10^24 kg"
+            },
+            {
+                "key": "Temperature(mean)",
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/solar_system.json]: Can you add a space before the `(` so it's `Temperature (mean)`

File: share/goodie/cheat_sheets/json/solar_system.json [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+            }
+        ],
+        "Uranus": [
+            {
+                "key": "Distance from Sun (minimum/maximum)",
+                "val": "2,748,938,461/3,004,419,704 km"
+            },
+            {
+                "key": "Mass",
+                "val": "(8.6810 ± 0.0013)×10^25 kg"
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/solar_system.json]: Is there a special reason why Uranus has defined error whilst the others don't? Personally I'd prefer error indications on all if available

File: share/goodie/cheat_sheets/json/solar_system.json (hunk 5 above)
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/solar_system.json]: Also I'm assuming can we make it clear that this is the mean surface temperature

File: share/goodie/cheat_sheets/json/solar_system.json (hunk 3 above)
Comment (vikashvverma) [inline share/goodie/cheat_sheets/json/solar_system.json]: People won't go to see the distance of earth from Sun in Sun section and vice versa.

This is just to make sure basic information are present in the respective sections.

File: share/goodie/cheat_sheets/json/solar_system.json (hunk 6 above)
Comment (vikashvverma) [inline share/goodie/cheat_sheets/json/solar_system.json]: @mintsoft It's available only for Uranus.

File: share/goodie/cheat_sheets/json/solar_system.json (hunk 5 above)
Comment (vikashvverma) [inline share/goodie/cheat_sheets/json/solar_system.json]: `mean surface temperature ` won't look good as title.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=solar system cheatsheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/solar_system_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/solar_system.json (hunk 5 above)
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/solar_system.json]: How about `Temperature (mean surface)` then?

Comment (vikashvverma) [inline share/goodie/cheat_sheets/json/solar_system.json]: I think `Surface Temperature (mean)` would be good.
`Temperature (surface mean)` or `Temperature (mean surface)` does not sound good.
What do you say?

Comment (mintsoft) [inline share/goodie/cheat_sheets/json/solar_system.json]: Yeah `Surface Temperature (mean)` works

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=solar system cheatsheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/solar_system_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/solar_system.json (hunk 3 above)
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/solar_system.json]: I agree with @mintsoft and it seems a bit inconsistent to have the rest in terms of distance to Sun, but then this as distance to Earth.

File: share/goodie/cheat_sheets/json/solar_system.json [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+        "Jupiter",
+        "Saturn",
+        "Uranus",
+        "Neptune"
+    ],
+    "sections": {
+        "Sun": [
+            {
+                "key": "Distance from earth",
+                "val": "149,597,900 km"
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/solar_system.json]: This is different to the `distance to Sun` present in the section on Earth.

File: share/goodie/cheat_sheets/json/solar_system.json [hunk 8]
```diff
@@ -0,0 +1,200 @@
+{
+    "id": "solar_system_cheat_sheet",
+    "name": "Solar System",
+    "description": "Basic information about our solar system",
+    "metadata": {
+        "sourceName": "Wikipedia",
+        "sourceUrl": "https://en.wikipedia.org/wiki/Solar_System"
+    },
+    "aliases": [
+        "Solar system",
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/solar_system.json]: Aliases should be lower-case.

**Comment by GuiltyDolphin** [comment]
@vikashvverma The file name should be `solar-system.json`. Please change the file name after you have resolved the current comments.

File: share/goodie/cheat_sheets/json/solar_system.json [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+            }
+        ],
+        "Mercury": [
+            {
+                "key": "Distance from Sun (minimum/maximum)",
+                "val": " 46,001,200/69,816,900 km"
+            },
+            {
+                "key": "Mass",
+                "val": "3.3011×10^23 kg"
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/solar_system.json]: @vikashvverma We can use Unicode!

Please replace all `^X` with the superscript equivalents.

File: share/goodie/cheat_sheets/json/solar_system.json (hunk 7 above)
Comment (vikashvverma) [inline share/goodie/cheat_sheets/json/solar_system.json]: That is mean distance.
In earth section it's minimum and maximum.

File: share/goodie/cheat_sheets/json/solar_system.json (hunk 9 above)
Comment (vikashvverma) [inline share/goodie/cheat_sheets/json/solar_system.json]: It has been resolved.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=solar system cheatsheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/solar_system_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=solar system cheatsheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/solar_system_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/solar-system.json [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+        "Mars",
+        "Jupiter",
+        "Saturn",
+        "Uranus",
+        "Neptune"
+    ],
+    "sections": {
+        "Sun": [
+            {
+                "key": "Distance from earth",
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/solar-system.json]: Perhaps we should use `Mean distance from Earth` then.

File: share/goodie/cheat_sheets/json/solar-system.json [hunk 11]
```diff
@@ -0,0 +1,200 @@
...
+            }
+        ],
+        "Venus": [
+            {
+                "key": "Distance from Sun (minimum/maximum)",
+                "val": "107,477,000/108,939,000 km"
+            },
+            {
+                "key": "Mass",
+                "val": "4.8675×10\u00B2\u2074 kg"
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/solar-system.json]: What's this? Use the actual Unicode symbols, like you have for the multiplication sign.

File: share/goodie/cheat_sheets/json/solar-system.json [hunk 12]
```diff
@@ -0,0 +1,200 @@
+{
+    "id": "solar_system_cheat_sheet",
+    "name": "Solar System",
+    "description": "Basic information about our solar system",
+    "metadata": {
+        "sourceName": "Wikipedia",
+        "sourceUrl": "https://en.wikipedia.org/wiki/Solar_System"
+    },
+    "aliases": [
+        "solar system",
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/solar-system.json]: We don't need this alias (it is dealt with by file name).

File: share/goodie/cheat_sheets/json/solar-system.json (hunk 10 above)
Comment (vikashvverma) [inline share/goodie/cheat_sheets/json/solar-system.json]: Fixed that.

Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/solar-system.json]: @vikashvverma It's not showing up for me.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=solar system cheatsheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/solar_system_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/solar-system.json (hunk 10 above)
Comment (vikashvverma) [inline share/goodie/cheat_sheets/json/solar-system.json]: It's there now.

File: share/goodie/cheat_sheets/json/solar-system.json [hunk 13]
```diff
@@ -0,0 +1,199 @@
+{
+    "id": "solar_system_cheat_sheet",
+    "name": "Solar System",
+    "description": "Basic information about our solar system",
+    "metadata": {
+        "sourceName": "Wikipedia",
+        "sourceUrl": "https://en.wikipedia.org/wiki/Solar_System"
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/solar-system.json]: How about using somewhere like http://nineplanets.org/ or https://solarsystem.nasa.gov/planets/ ? 

I prefer to stay away from Wikipedia for cheat sheets seeing as [ordinary queries](https://duckduckgo.com/?q=solar+system&t=ffab&ia=about) will trigger it.

Comment (vikashvverma) [inline share/goodie/cheat_sheets/json/solar-system.json]: I thought about that.

But the information source was wikipedia.

Should I update the source.
What do you say?

File: share/goodie/cheat_sheets/json/solar-system.json [hunk 14]
```diff
@@ -0,0 +1,199 @@
...
+                "key": "Mass",
+                "val": "1.989 x 10³⁰ kg"
+            },
+            {
+                "key": "Temperature (core)",
+                "val": "1.57×10⁷ K"
+            },
+            {
+                "key": "Volume",
+                "val": "1.41×10¹⁸ km³"
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/solar-system.json]: This one isn't displaying correctly for me:

screenshot from 2016-02-07 10 37 03

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=solar system cheatsheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/solar_system_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=solar system cheatsheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/solar_system_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/solar-system.json (hunk 14 above)
Comment (vikashvverma) [inline share/goodie/cheat_sheets/json/solar-system.json]: It's proper now.
That was fixed in second last commit.

File: share/goodie/cheat_sheets/json/solar-system.json (hunk 13 above)
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/solar-system.json]: I'd prefer a switch.

You can do a quick check of the info from the other sites anyway, for example, on `nineplanets`, for Jupiter it gives:

screenshot from 2016-02-07 10 39 56

Comment (vikashvverma) [inline share/goodie/cheat_sheets/json/solar-system.json]: Sure.

File: share/goodie/cheat_sheets/json/solar-system.json (hunk 14 above)
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/solar-system.json]: Right you are. I'll wait a bit for Beta to update before commenting on style ;) I think Dax needs his coffee.

File: share/goodie/cheat_sheets/json/solar-system.json (hunk 13 above)
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/solar-system.json]: Nine planets provides a [summary of data](http://nineplanets.org/data.html) as well if you're interested.

Comment (vikashvverma) [inline share/goodie/cheat_sheets/json/solar-system.json]: Temperature data are not consistent (or unavailable) on this site

In this case I would have to remove this from the cheat sheet.

Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/solar-system.json]: @vikashvverma There are other sites you can use - that is just a suggestion.

And you don't have to stick *rigidly* to the data source - if you find multiple sources agreeing upon a value you can use that - have you checked out the NASA source?

Comment (vikashvverma) [inline share/goodie/cheat_sheets/json/solar-system.json]: Updated the cheat sheet and data source as well.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=solar system cheatsheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/solar_system_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=solar system cheatsheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/solar_system_cheat_sheet_mobile.png)

**Comment by GuiltyDolphin** [comment]
@moollaza This is looking pretty good to me, although @mintsoft is probably quite a bit more qualified in the topic than me :wink:

**Comment by moollaza** [comment]
This LGTM :+1: 

I think we should split the min/max dist from the sun into two different key-val pairs, or at least put a space around the `/` to improve the legibility. It's a little hard for me to read the numbers there as-is.

**Comment by vikashvverma** [comment]
@moollaza I have created PR https://github.com/duckduckgo/zeroclickinfo-goodies/pull/2342 to add spaces.

**Pull request merged**